Maple2.Server.Game/Manager/GameEventUserValueManager.cs

```diff
@@ -25,32 +26,44 @@ public sealed class GameEventUserValueManager {
         eventValues = db.GetEventUserValues(session.CharacterId);
     }
 
-    public object this[GameEventUserValueType type] {
-        set {
-            if (!eventValues.TryGetValue(type, out GameEventUserValue? gameEventUserValue)) {
-                throw new ArgumentOutOfRangeException(nameof(type), type, "Invalid event value type.");
-            }
-            gameEventUserValue.Value = value.ToString() ?? throw new InvalidOperationException("Invalid new value");
-            session.Send(GameEventUserValuePacket.Update(gameEventUserValue));
+    public void Set(GameEventUserValueType type, int gameEventId, object value) {
```

Comment:
I would rearrange the parameter order to have `eventId` first since that's the top-level scope